### PR TITLE
Setup modal for creating task

### PR DESCRIPTION
**Pull request by hopeaisulu**

### CODE REVIEW

**Review by TilegenGena — APPROVED** [review]

File: server/src/modules/task/task.controller.ts [hunk 1]
```diff
@@ -17,15 +17,15 @@ export class TaskController {
...
+    return this.taskService.getAllTasks();
   }
 
   @Get(':id')
   async getTaskById(@Param('id') id: number): Promise<Task> {
     return this.taskService.getTaskById(id);
   }
 
-  @Post('create')
+  @Post('task')
```
Comment (TilegenGena) [inline server/src/modules/task/task.controller.ts:28]: When will creating, link gonna  be `***/task/task`

Comment (TilegenGena) [inline server/src/modules/task/task.controller.ts:28]: When will creating, link gonna  be `***/task/task`?

**Review by TilegenGena — APPROVED** [review]

**Review by TilegenGena — APPROVED** [review]

**Review by TilegenGena — APPROVED** [review]

File: server/src/modules/task/task.controller.ts (hunk 1 above)
Comment (hopeaisulu) [reply server/src/modules/task/task.controller.ts:28]: This should be changed to something else, but not to task.

Comment (networkflow) [reply server/src/modules/task/task.controller.ts:28]: You could just do `@Post('')`, so the link will be POST `***/task/`

File: client/angular.json [hunk 2]
```diff
@@ -40,13 +35,13 @@
               "budgets": [
                 {
                   "type": "initial",
-                  "maximumWarning": "500kb",
-                  "maximumError": "1mb"
+                  "maximumWarning": "5mb",
```
Comment (networkflow) [inline client/angular.json:38]: Haha this is probably good, I kept hitting the 500kb error on the tutorial app!

Comment (networkflow) [reply client/angular.json:38]: (btw nit: if you're making a change like this that you didn't explain in the commit message and that's mixed in with a lot of other changes due to installing `ng-bootstrap`, it's easy for reviewers to miss. I would recommend either mentioning it in the commit message or adding an inline comment on the code in GitHub to mention what you're doing here.)

File: client/src/app/components/task-modal/task-modal/task-modal.component.html [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      formControlName="title"
+    />
+    <div
+      *ngIf="
+        taskForm.controls['title'].invalid && taskForm.controls['title'].touched
+      "
+      class="alert alert-danger"
+    >
+      Title is required.
+    </div>
```
Comment (networkflow) [inline client/src/app/components/task-modal/task-modal/task-modal.component.html:26]: This seems neat! Not too familiar with `ng-bootstrap` but it looks good to me

File: client/src/app/components/task-modal/task-modal/task-modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import { NgbActiveModal, NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ITask } from 'src/app/interfaces/task';
+
+@Component({
+  selector: 'app-task-modal',
+  templateUrl: './task-modal.component.html',
+  styleUrls: ['./task-modal.component.scss'],
+})
+export class TaskModalComponent {
+  @Input() task!: ITask;
```
Comment (networkflow) [inline client/src/app/components/task-modal/task-modal/task-modal.component.ts:12]: I'm a little confused by `task!: ITask`: should it be `task: ITask | null`? I'm assuming that a missing task means it's a create modal, but if there's a task it's for editing.

We definitely want TypeScript to know that `task` can be null/undefined. I'm not super familiar with the `task!: ITask` syntax: does TypeScript know that `task` can be null/undefined, and you're just saying that the type must be `ITask` if the input is present?

File: client/src/app/interfaces/task.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export interface ITask {
```
Comment (networkflow) [inline client/src/app/interfaces/task.ts:1]: If you don't want to use a `shared` folder (which is fine), I would at least add a comment to both the server/client `ITask` files saying to make sure to keep these in sync!

File: client/src/app/interfaces/task.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  id: number;
+  title: string;
+  menteeId: number;
+  description: string;
+  status: string;
+  dueDate: Date;
```
Comment (networkflow) [inline client/src/app/interfaces/task.ts:7]: Already commented on the server-side commit, but I don't think you can send a JavaScript `Date` object to the server? Am I wrong?

**Review by networkflow — COMMENTED** [review]
Overall looks great to me, nice work! Added some suggestions

**Pull request merged**